opal/threads/mutex.h

```diff
@@ -26,167 +26,6 @@
- * process, return immediately.
- */
-#define OPAL_THREAD_LOCK(mutex)                 \
-    do {                                        \
-        if (OPAL_UNLIKELY(opal_using_threads())) {      \
-            opal_mutex_lock(mutex);             \
-        }                                       \
-    } while (0)
-
-
-/**
- * Try to lock a mutex if opal_using_threads() says that multiple
- * threads may be active in the process.
- *
- * @param mutex Pointer to a opal_mutex_t to trylock
- *
- * If there is a possibility that multiple threads are running in the
- * process (as determined by opal_using_threads()), this function will
- * trylock the mutex.
- *
- * If there is no possibility that multiple threads are running in the
- * process, return immediately without modifying the mutex.
- *
- * Returns 0 if mutex was locked, non-zero otherwise.
- */
-#define OPAL_THREAD_TRYLOCK(mutex)                      \
-    (OPAL_UNLIKELY(opal_using_threads()) ? opal_mutex_trylock(mutex) : 0)
-
-/**
- * Unlock a mutex if opal_using_threads() says that multiple threads
- * may be active in the process.
- *
- * @param mutex Pointer to a opal_mutex_t to unlock.
- *
- * If there is a possibility that multiple threads are running in the
- * process (as determined by opal_using_threads()), this function will
- * unlock the mutex.
- *
- * If there is no possibility that multiple threads are running in the
- * process, return immediately without modifying the mutex.
- */
-#define OPAL_THREAD_UNLOCK(mutex)               \
-    do {                                        \
-        if (OPAL_UNLIKELY(opal_using_threads())) {      \
-            opal_mutex_unlock(mutex);           \
-        }                                       \
-    } while (0)
-
-
-/**
- * Lock a mutex if opal_using_threads() says that multiple threads may
- * be active in the process for the duration of the specified action.
- *
- * @param mutex    Pointer to a opal_mutex_t to lock.
- * @param action   A scope over which the lock is held.
- *
- * If there is a possibility that multiple threads are running in the
- * process (as determined by opal_using_threads()), this function will
- * acquire the lock before invoking the specified action and release
- * it on return.
- *
- * If there is no possibility that multiple threads are running in the
- * process, invoke the action without acquiring the lock.
- */
-#define OPAL_THREAD_SCOPED_LOCK(mutex, action)  \
-    do {                                        \
-        if(OPAL_UNLIKELY(opal_using_threads())) {       \
-            opal_mutex_lock(mutex);             \
-            action;                             \
-            opal_mutex_unlock(mutex);           \
-        } else {                                \
-            action;                             \
-        }                                       \
-    } while (0)
-
-END_C_DECLS
+#include "opal/mca/threads/mutex.h"
```

Comment:
Ditto with `opal/threads/condition.h`; it looks like this file should be removed.